include/revng/Model/Type.h

```diff
@@ -24,8 +24,6 @@ name: Type
 doc: Base class of model types used for LLVM-style RTTI
 type: struct
 fields:
-  - name: Kind
-    type: model::TypeKind::Values
```

Comment:
Some observations:
* It is nice that we can drop this, but it makes things a bit magic for someone passing by.
However, even if we were to force the user to put by hand the `Kind` field... what type would it have?
* Also, below `Kind` is listed as a part of the key, we're really forcing the user to be aware that if we have `abstract: true`, there's going to be a magic field `Kind`.
* Other question: does it make sense to have a `struct` where `Kind` is not part of the key?
  We have an invariant that says that inheritors must have the same key as the base class.
  I guess it's probably sane to have it, but not strictly necessary.

Overall, I think that if `abstract: true` we should ask the user to have a field named kind whose type is `${STRUCT_NAME}Kind`. If it's not there, we should print a nice error message that instructs him on what to do.
`${STRUCT_NAME}Kind` would be autogenerated.

What's your take?

Reply:
Yeah, documentation is always good, but I think that letting the user make a small mistake and give a very informative error on how to fix it, is also effective.

Reply:
Ok, I see your point.